core/arch/arm/kernel/boot.c

```diff
@@ -1423,6 +1423,29 @@ static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
 	init_sec_mon(nsec_entry);
 }
 
+#ifdef CFG_SECURE_PARTITION
+static void check_crc32_feature(void)
+{
+	static bool checked = false;
+	static bool available = false;
+
+	/* Check once (i.e. on the boot core) and store the result */
+	if (!checked) {
+		available = feat_crc32_implemented();
+		checked = true;
+		return;
+	}
+
+	/* If available on the boot core, make sure other cores match */
+	if (available) {
+		if (!feat_crc32_implemented()) {
+			EMSG("CRC32 not available on core %ld", get_core_pos());
+			panic();
+		}
+	}
+}
```

Comment:
```C
#else
static void check_crc32_feature(void)
{}
```
and remove the ifdefs in `boot_init_primary_late()` and `init_secondary_helper()`

Reply:
Done